kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -96,6 +105,56 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
     ///@name Operations
     ///@{
 
+    int Check(const ModelPart& rModelPart) const override
+    {
+        KRATOS_TRY
+
+        auto lambda2_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                rExtensions.GetFirstDerivativesVariables(rVec);
+            });
+        auto lambda3_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                return rExtensions.GetSecondDerivativesVariables(rVec);
+            });
+        auto auxiliary_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                return rExtensions.GetAuxiliaryVariables(rVec);
```

Comment:
why don't you directly get the names instead?

Reply:
Good question, I have no clue to be honest. May be we can change it in a future PR? @MFusseder ? For fluids, I can live with names only. :) @marcnunezc ?